modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/exportimport/data/handler/AssetListEntryAssetEntryRelStagedModelDataHandler.java

```diff
@@ -176,6 +176,10 @@ protected void doImportStagedModel(
 					portletDataContext, importedAssetListEntryAssetEntryRel);
 		}
 
+		if (existingAssetListEntryAssetEntryRel == null) {
+			return;
```

Comment:
I'm thinking we shouldn't quietly fail, we should at least throw a log or maybe some warning to the screen. Have you looked into what other components may be doing when a reference is missing? Alternatively, we could ask the team we got this component from (Echo right?)

Reply:
importedAssetListEntryAssetEntryRel == null ?

Reply:
also, in AssetDisplayPageStagedModelDataHandler there is this. i dont know what we might be able to copy from here, but it could be helpful to at least add some logging

```
if ((existingAssetDisplayPageEntry == null) ||
			!portletDataContext.isDataStrategyMirror()) {

			importedAssetDisplayPageEntry.setClassPK(existingClassPK);

			try {
				importedAssetDisplayPageEntry =
					_stagedModelRepository.addStagedModel(
						portletDataContext, importedAssetDisplayPageEntry);
			}
			catch (Exception exception) {
				if (_log.isWarnEnabled()) {
					_log.warn("Unable to add asset display page", exception);
				}

				portletDataContext.removePrimaryKey(
					ExportImportPathUtil.getModelPath(assetDisplayPageEntry));

				return;
			}
		}
		else {...
```

Reply:
The code you shared is incorrect. The problem is that the assetEntryPage can be imported before the entry. This shouldn't be fixed that way. Instead, the code should be modified to add a reference to entry when assetEntryPage is exported and import the references before start to import the assetEntryPages. This way, when assetEntryPage is imported, it will wait until entry is imported.

Reply:
At least it was the reason why he committed this change.